gp-includes/things/translation-set.php

```diff
@@ -31,6 +31,29 @@ public function restrict_fields( $set ) {
 		$set->project_id_should_not_be('empty');
 	}
 
+	// Field handling
+
+	public function normalize_fields( $args ) {
+		$args = (array)$args;
+
+		if ( isset( $args['name'] ) && empty( $args['name'] ) ) {
+			if ( isset( $args['locale'] ) && ! empty( $args['locale'] ) ) {
+				$locale = GP_locales::by_slug( $args['locale'] );
+				$args['name'] = $locale->english_name;
+			}
+		}
+		
+		if ( isset( $args['slug'] ) && !$args['slug'] ) {
+			$args['slug'] = $args['name'];
```

Comment:
That's why I used the english_name of the locale, so in your example it would be `German` as the slug.